jib-cli/src/main/java/com/google/cloud/tools/jib/cli/cli2/CacheDirectories.java

```diff
@@ -35,12 +37,15 @@
   /**
    * Create a caches helper for cli cache locations.
    *
-   * @param commonCliOptions cli options
-   * @param contextRoot the context root, if a single file, use the parent directory, this context
+   * @param commonCliOptions cli options for user configured cache directories
+   * @param contextRoot the context root, use the parent directory of single files, this context
    *     root must exist
    * @return an instance of CacheDirectories with cli specific cache locations
    */
   public static CacheDirectories from(CommonCliOptions commonCliOptions, Path contextRoot) {
+    Preconditions.checkArgument(
+        Files.isDirectory(contextRoot),
+        "contextRoot must be a directory, but " + contextRoot.toString() + " is not.");
```

Comment:
Just to confirm again: even if `Path` is a relative path (for example, `.`), `path.toFile().getCanonicalPath()` is always a full absolute path based on the current working directory?

Reply:
canonical path will return a the same path for the same file -- no matter what the currently relative pathing is and whatever the working directory is